core/src/main/java/io/temporal/samples/hello/HelloAccumulator.java

```diff
@@ -0,0 +1,619 @@
+
+/*
+ * Sample Temporal Workflow Definition that accumulates events.
+ * This sample implements the Accumulator Pattern: collect many meaningful
+ *  things that need to be collected and worked on together, such as
+ *  all payments for an account, or all account updates by account.
+ *
+ * This sample models robots being created throughout the time period,
+ *  groups them by what color they are, and greets all the robots
+ *  of a color at the end.
+ *
+ * A new workflow is created per grouping. Workflows continue as new as needed.
+ *  A sample activity at the end is given, and you could add an activity to
+ *  process individual events in the
+ */
+public class HelloAccumulator {
+  // set a time to wait for another signal to come in, e.g.
+  //    Duration.ofDays(30);
+  static final Duration MAX_AWAIT_TIME = Duration.ofMinutes(1);
+
+  static final String TASK_QUEUE = "HelloAccumulatorTaskQueue";
+  static final String WORKFLOW_ID_PREFIX = "HelloAccumulatorWorkflow";
+
+  public static class Greeting implements Serializable {
+    String greetingText;
+    String bucket;
+    String greetingKey;
+
+    public String getGreetingText() {
+      return greetingText;
+    }
+
+    public void setGreetingText(String greetingText) {
+      this.greetingText = greetingText;
+    }
+
+    public String getBucket() {
+      return bucket;
+    }
+
+    public void setBucket(String bucket) {
+      this.bucket = bucket;
+    }
+
+    public String getGreetingKey() {
+      return greetingKey;
+    }
+
+    public void setGreetingKey(String greetingKey) {
+      this.greetingKey = greetingKey;
+    }
+
+    public Greeting(String greetingText, String bucket, String greetingKey) {
+      this.greetingText = greetingText;
+      this.bucket = bucket;
+      this.greetingKey = greetingKey;
+    }
+
+    public Greeting() {}
+
+    @Override
+    public String toString() {
+      return "Greeting [greetingText="
+          + greetingText
+          + ", bucket="
+          + bucket
+          + ", greetingKey="
+          + greetingKey
+          + "]";
+    }
+  }
+
+  /**
+   * The Workflow Definition's Interface must contain one method annotated with @WorkflowMethod.
+   *
+   * <p>Workflow Definitions should not contain any heavyweight computations, non-deterministic
+   * code, network calls, database operations, etc. Those things should be handled by the
+   * Activities.
+   *
+   * @see io.temporal.workflow.WorkflowInterface
+   * @see io.temporal.workflow.WorkflowMethod
+   */
+  @WorkflowInterface
+  public interface AccumulatorWorkflow {
+    /**
+     * This is the method that is executed when the Workflow Execution is started. The Workflow
+     * Execution completes when this method finishes execution.
+     */
+    @WorkflowMethod
+    String accumulateGreetings(
+        String bucketKey, Deque<Greeting> greetings, Set<String> allGreetingsSet);
+
+    // Define the workflow waitForName signal method. This method is executed when
```

Comment:
Do you need the first sentence at all?  If so, I think `waitForName` is an older name?

Reply:
good call, changed.